app/src/main/java/com/example/tasky/feature_agenda/data/repository/AgendaRepositoryImpl.kt

```diff
@@ -2,114 +2,161 @@ package com.example.tasky.feature_agenda.data.repository
-            val localTasks = daos.taskDao.getTasks().map { it.toTask() }
-            val localReminders = daos.reminderDao.getReminders().map { it.toReminder() }
+            val localEvents = eventWithAttendees.map { it.toEvent() }
+            val localReminders = reminders.map { it.toReminder() }
+            val localTasks = tasks.map { it.toTask() }
+
+            (localEvents + localReminders + localTasks).sortedBy { it.sortDate }
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override suspend fun fetchAgendaFromRemote(): AgendaResult {
 
-            val localAgenda = localEvents + localTasks + localReminders
+        val remoteAgenda = try {
+            val response = api.getAgenda(
+                timeZone = TimeZone.getDefault().toString(),
+                time = System.currentTimeMillis()
+            )
 
-            emit(Resource.Success(data = localAgenda))
+            val remoteEvents = response.body()?.events?.map { it.toEvent() } ?: emptyList()
+            val remoteReminders = response.body()?.reminders?.map { it.toReminder() } ?: emptyList()
+            val remoteTasks = response.body()?.tasks?.map { it.toTask() } ?: emptyList()
 
-            val isDbEmpty =
-                localEvents.isEmpty() && localTasks.isEmpty() && localReminders.isEmpty()
-            val shouldJustLoadFromCache = !isDbEmpty && !fetchFromRemote
+            remoteEvents + remoteTasks + remoteReminders
 
-            if (shouldJustLoadFromCache) {
-                emit(Resource.Loading(false))
-                return@flow
-            }
+        } catch (e: HttpException) {
+            return AgendaResult.Error(e.message ?: "Invalid Response")
+        } catch (e: IOException) {
+            return AgendaResult.Error(e.message ?: "Couldn't reach server")
+        }
+
+        remoteAgenda.let { agenda ->
+            db.eventDao.deleteEvents()
+            db.eventDao.deleteAttendees()
+            db.reminderDao.deleteReminders()
+            db.taskDao.deleteTasks()
 
-            val remoteAgenda = try {
-                val response = api.getAgenda(
-                    timeZone = TimeZone.getDefault().toString(),
-                    time = System.currentTimeMillis()
-                )
+            val eventList = agenda.filterIsInstance<AgendaItem.Event>()
+            val remindersList = agenda.filterIsInstance<AgendaItem.Reminder>()
+            val tasksList = agenda.filterIsInstance<AgendaItem.Task>()
 
-                val remoteEvents = response.body()?.events?.map { it.toEvent() } ?: emptyList()
-                val remoteTasks = response.body()?.tasks?.map { it.toTask() } ?: emptyList()
-                val remoteReminders = response.body()?.reminders?.map { it.toReminder() } ?: emptyList()
+            eventList.forEach { event->
+                db.eventDao.upsertEvent(event.toEventEntity())
+                db.eventDao.upsertAttendee(event.attendees.map { it.toAttendeeEntity(event.eventId) })
```

Comment:
I personally find it much more readable to extract such lambdas into a separate variable. Ideally, code should read like a book (meaning, you don't need to go through a lot of logic in your mind to understand what code does). This line is not immediately clear, since you have to evaluate the result of the `map` function in your head. By simply extracting it into a separate variable, you can make it much clearer:
```
val attendeeEntities = event.attendees.map { it.toAttendeeEntity(event.eventId) }
db.eventDao.upsertAttendee(attendeeEntities)
```
That way, your brain can skip evaluating the `map` function, since it just needs to read `attendeeEntities` to understand what the result represents 😄